### PR TITLE
[backend](test) add composer mock and test mutations

### PR DESCRIPTION
**Pull request by maelv-filigran**
### Proposed changes

* Added comprehensive integration tests for Connector Composer and Managed Connectors functionality
* Implemented XTMComposerMock to simulate the XTM Composer container orchestrator behavior
* Added test coverage for the complete connector lifecycle: Create/Deploy, Start, Stop, Delete
* Included tests for GraphQL mutations and queries related to connector management
* Added permission checks to ensure proper access control for connector operations
* Implemented error handling tests for edge cases and failure scenarios
* Added concurrent operation tests to validate system behavior under load

### Related issues
* #7328 
* Harmonize existing connector management with product specification
* Add integration tests for connector lifecycle management

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

This PR adds integration tests that mock the XTM Composer responses to validate the complete connector management flow. The tests cover:

1. **Connector Composer operations**: Registration, updates, status management, and queries
2. **Managed Connector lifecycle**: Creation, deployment, start/stop operations, and deletion
3. **XTM Composer GraphQL integration**: Tests for updateConnectorLogs, updateConnectorCurrentStatus, and connectorsForManagers
4. **Permission validation**: Ensures only authorized users can perform sensitive operations
5. **Error handling**: Validates graceful handling of failures and missing configurations
6. **Concurrent operations**: Tests system behavior when multiple connectors are managed simultaneously

The XTMComposerMock implementation provides a realistic simulation of the container orchestrator, allowing us to test the integration without requiring the actual XTM Composer service. This approach ensures reliable and fast test execution while maintaining comprehensive coverage of the connector management flow.

### CODE REVIEW

**Comment by natalie-natsu** [comment]
Hey @maelv-filigran, Nice Job! Can you target your PR on release/current? :-) PRs on main (master) are only for fixes and patch releases.

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/connector-composer-test.ts [hunk 1]
```diff
@@ -0,0 +1,931 @@
...
+    it('should update connector composer status', async () => {
+      const previousResult = await queryAsAdminWithSuccess({
+        query: CONNECTOR_MANAGER_QUERY,
+        variables: { managerId: TEST_COMPOSER_ID }
+      });
+      expect(previousResult.data).toBeDefined();
+      const previousSync = previousResult.data?.connectorManager.last_sync_execution;
+
+      // Wait a bit to ensure timestamp difference
+      await new Promise((resolve) => {
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/connector-composer-test.ts]: As side note, there is a method 'waitInSec' and 'wait' here https://github.com/OpenCTI-Platform/opencti/blob/3298365a5a8e65850ae1befd771a3f6de977464a/opencti-platform/opencti-graphql/src/database/utils.js#L333 that I think is similar.

Comment (maelv-filigran) [reply opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/connector-composer-test.ts]: Great! I'll use it!

**Review by aHenryJard — APPROVED** [review]
That's great job, thanks !
